code/model/DMSDocument.php

```diff
@@ -1237,6 +1244,7 @@ class DMSDocument_Controller extends Controller
 {
 
     public static $testMode = false;   //mode to switch for testing. Does not return document download, just document URL
+    public static $downloadBehavior = 'download';
```

Comment:
You've added a new feature :)

This should use the config system and so should be `private static` and it should also be on the model instead of the controller.